python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -283,6 +288,17 @@ def _task_done_callback(task_index, inputs):
             self._finished_streaming_gens.append(task.get_waitable())
             # Notify output queue that this task is complete.
             self._output_queue.notify_task_completed(task_index)
+            # Update estimate for blocks outputted
+            self._num_tasks_finished += 1
+            self._num_output_blocks += task.get_num_output_blocks()
+            num_tasks = 1
+            if len(self.input_dependencies) == 1:
+                # The number of outputs reported by upstream operator. We later
+                #  update self.num_outputs_total() with this estimate
+                num_tasks = self.input_dependencies[0].num_outputs_total()
```

Comment:
I just realized that this is still not accurate.  Because for "map_batches", we need to bundle the inputs to match the batch size. Thus, one input != one task. A better estimation of total # of tasks would be : `self.input_dependencies[0].num_outputs_total() / num_inputs_received * self._num_tasks_finished`.
For `num_inputs_received`, we just increment it in `add_input`.

Reply:
This estimate is good when the backpressure works well. If a bunch of inputs pile up, `_num_tasks_finished` lags behind `num_inputs_received` and the estimated num_tasks is too small. If I run balaji's example locally the estimate pretty much grows linearly along with the progress, so there's not really a difference. Is this okay?

Reply:
We could also use the ratio of `# of calls to add_input()` to `# of calls to _add_bundled_input()` for the estimate

Reply:
Good point!

Reply:
> use the ratio of # of calls to add_input() to # of calls to _add_bundled_input()
 
This makes more sense.